### PR TITLE
Fix gallery ssr issues [ch29227]

### PR DESCRIPTION
**Pull request by bstst**
https://app.clubhouse.io/cgtrader/story/29227/fix-gallery-ssr-issues

### CODE REVIEW

**Comment by bstst** [comment]
Sorry for this review being so large.

I had to re-introduce a simplified lib packaging solution for this to start working reliably on the SSR.

I also had to remove `aphrodite` which was used for styling and extract all the relevant styles and introduce css modules which extract their styles to the `lib/Lightbox.css` file, which in turn is imported in to the relevant CGT scss files.

**Review by andrejikonikov — APPROVED** [review]

File: src/components/Arrow.js [hunk 1]
```diff
@@ -16,15 +16,19 @@ function Arrow ({
...
-
 	return (
 		<button
 			type="button"
-			className={css(classes.arrow, classes['arrow__direction__' + direction], size && classes['arrow__size__' + size])}
+			className={`arrow ${styles.arrow} ${styles[`arrow__size__${size}`]}`}
 			onClick={onClick}
 			onTouchEnd={onClick}
-			{...props}
+			style={{
```
Comment (RokasC) [inline src/components/Arrow.js:25]: Why have this here and not in css ?

File: src/components/Thumbnail.js [hunk 2]
```diff
@@ -1,41 +1,28 @@
...
-
 	const url = type === 'marmoset' ? marmosetThumb : thumbnail ? thumbnail : src;
-	const classes = StyleSheet.create(deepMerge(defaultStyles, theme));
 
 	return (
 		<div
-			className={css(classes.thumbnail, active && classes.thumbnail__active)}
+			className={`thumbnail ${active ? 'thumbnail-active' : ''} ${styles.thumbnail}`}
+			style={{
+				backgroundImage: 'url("' + url + '")',
```
Comment (RokasC) [inline src/components/Thumbnail.js:15]: Could use string interpolation here. `url('${url}')` would look cleaner IMHO

File: src/components/Video.js [hunk 3]
```diff
@@ -14,7 +15,8 @@ class Video extends React.Component {
 	}
 
 	loadPlayer () {
-		require(['react-player'], (mod) => this.setState({ VideoComponent: mod.default }));
+		window.React = React;
+		loadScript('https://unpkg.com/react-player@1.11.0/dist/ReactPlayer.js').then(() => this.setState({ loaded: true }));
```
Comment (RokasC) [inline src/components/Video.js:19]: Do we really need to depend on external URL? Can't load it via npm?

File: src/components/Arrow.js (hunk 1 above)
Comment (bstst) [reply src/components/Arrow.js:25]: These are remnants of the _old aphrodite way_, they are dynamically overridable by providing a `theme={theme}` object to the gallery.

Of course they may be factored out, but I suggest scoping it out to another, let's call it a _cleanup_ story if need be.

File: src/components/Video.js (hunk 3 above)
Comment (bstst) [reply src/components/Video.js:19]: There are major issues making libraries load additional code asynchronously. Loading it using npm like it was done before (`require(['react-player'])`) introduces code-split chunks to the library bundled separately, but webpack doesn't know where to load these async assets from when they need to be loaded.

And if **sokra** says it's not yet supported — https://github.com/webpack/webpack/issues/6818#issuecomment-377225847 — then it's not yet supported.

I shall investigate our options here though, but can't promise anything.

File: src/components/Thumbnail.js (hunk 2 above)
Comment (bstst) [reply src/components/Thumbnail.js:15]: This is a remnant of the old code, so just a one-of-many copy-pastes. There are a lot of possible improvements to the library code (convert tabs to spaces among them), they may be done separately.

**Review by RokasC — APPROVED** [review]

**Pull request merged**